apps/system/js/notification_screen.js

```diff
@@ -110,20 +106,20 @@ var NotificationScreen = {
 
     var self = this;
     SettingsListener.observe('notification.ringtone', '', function(value) {
-      self._sound = self.ringtoneURL.set(value);
+      LazyLoader.load(['shared/js/settings_url.js']).then(function() {
+        self.ringtoneURL = new SettingsURL();
+        self._sound = self.ringtoneURL.set(value);
+      });
     });
 
     // We have new default ringtones in 2.0, so check if the version is upgraded
     // then execute the necessary migration.
-    VersionHelper.getVersionInfo().then(function(versionInfo) {
-      if (versionInfo.isUpgrade()) {
-        LazyLoader.load('js/tone_upgrader.js', function() {
-          toneUpgrader.perform('alerttone');
-        });
-      }
-    }, function(err) {
-      console.error('VersionHelper failed to lookup version settings.');
-    });
+    if (Service.query('justUpgraded')) {
+      LazyLoader.load('js/tone_upgrader.js', function() {
+        toneUpgrader.perform('alerttone');
+      });
+    }
+    Service.register('clearAll', this);
```

Comment:
Should this be called `clearAllNotifications`? Who else would consume this service besides the NotificationScreen itself?